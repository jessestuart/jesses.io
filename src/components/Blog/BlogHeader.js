/* @flow */
import { Calendar } from 'react-feather'
import React, { Fragment } from 'react'
import _ from 'lodash'

import classNames from 'classnames'
import voca from 'voca'

import StyledLink from '../StyledLink/StyledLink'

type Props = {
  className?: string,
  date?: string,
  children: string,
  link?: string,
  location: {
    pathname: string,
  },
}

const BlogHeader = ({ children, className, date, link, location }: Props) => {
  // If we pass in a link, just href to that. Otherwise generate a
  // `/path#section-header`-style ID for e.g., linking to section headers.
  const pathname = _.get(location, 'pathname')
  const headerID = _.isEmpty(link) ? voca.slugify(children) : ''
  const headerLink = _.isEmpty(pathname) ? `/#${headerID}` : link

  return (
    <Fragment>
      <h2
        className={classNames(
          className,
          'alegreya-sans f2 fw5 header-primary mt4'
        )}
        id={headerID}
      >
        <StyledLink href={headerLink}>{children}</StyledLink>
      </h2>
      {_.isNil(date) ? null : (
        <p className="f6 mv2 flex items-center">
          <Calendar
            className="header-primary mr2"
            style={{ maxWidth: '15px' }}
          />
          {date}
        </p>
      )}
    </Fragment>
  )
}

export default BlogHeader
