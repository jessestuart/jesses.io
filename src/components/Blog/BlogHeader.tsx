import classNames from 'classnames'
import _ from 'lodash'
import React from 'react'
import { Calendar } from 'react-feather'
import voca from 'voca'

import StyledLink from '../StyledLink/StyledLink'

interface Props {
  className?: string
  date?: string
  children: string
  link?: string
  // @Deprecated (?)
  location?: {
    pathname: string
  }
  title?: string
}

const BlogHeader = ({
  children,
  className,
  date,
  link,
  title,
  location,
}: Props) => {
  // If we pass in a link, just href to that. Otherwise generate a
  // `/path#section-header`-style ID for e.g., linking to section headers.
  // const pathname = _.get(location, 'pathname')
  const headerID = _.isEmpty(link) ? voca.slugify(title) : link
  const headerLink = link || `/${location.pathname}#${headerID}`

  return (
    <>
      <h2
        className={classNames(
          className,
          'alegreya-sans f2 fw5 header-primary mt4',
        )}
        id={headerID}
      >
        <StyledLink href={headerLink}>{children}</StyledLink>
      </h2>
      {date && (
        <p className="f6 mv2 flex items-center">
          <Calendar
            className="header-primary mr2"
            style={{ maxWidth: '15px' }}
          />
          {date}
        </p>
      )}
    </>
  )
}

export default BlogHeader
