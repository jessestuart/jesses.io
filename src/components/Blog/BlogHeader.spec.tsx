import React from 'react'
import renderer from 'react-test-renderer'
import BlogHeader from './BlogHeader'

test('Render BlogHeader component.', () => {
  const dateISO = '2018-01-01'
  const tree = renderer
    .create(
      <BlogHeader date={dateISO} slug={dateISO}>
        FooBar
      </BlogHeader>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
