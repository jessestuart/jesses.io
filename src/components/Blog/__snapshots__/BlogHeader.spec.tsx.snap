// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BlogHeader component. 1`] = `
Array [
  <h2
    className="alegreya-sans f2 fw5 header-primary mt4"
    id="foo-bar"
  >
    <a
      className="sc-bdVaJa iwQflW"
      href="undefined#foo-bar"
    >
      FooBar
    </a>
  </h2>,
  <p
    className="f6 mv2 flex items-center"
  >
    <svg
      className="header-primary mr2"
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "maxWidth": "15px",
        }
      }
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="18"
        rx="2"
        ry="2"
        width="18"
        x="3"
        y="4"
      />
      <line
        x1="16"
        x2="16"
        y1="2"
        y2="6"
      />
      <line
        x1="8"
        x2="8"
        y1="2"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="10"
        y2="10"
      />
    </svg>
    2018-01-01
  </p>,
]
`;
